watershed_workflow/condition.py

```diff
@@ -631,3 +632,13 @@ def bank_nodes_from_edge(edge, elem, m2):
     bank_tri = cells_to_edge[0]
     node_id = (set(bank_tri) - set(edge)).pop()
     return node_id
+
+
+def elevate_rivers(rivers, crs, dem, dem_profile):
+    """elevate river using dem and store reach-bed-profile as node properties"""
+    for river in rivers:
+        for i, node in enumerate(river.preOrder()):
+                node_points=(np.array(node.segment.xy).T)
+                node_elevs = watershed_workflow.elevate(node_points, crs, dem, dem_profile)[:,2] 
```

Comment:
So I really don't want (wherever possible) for "low level" API to call "high level" API.

We should refactor (and we can do this post-PR) to move the "elevate" function into low-level API and call that directly from here.  The high-level API is meant to only be there for users to group common usages of capability.  This would also fix your logging problem (see commented logging in hilev.elevate() ).

Reply:
I agree. I wanted to use the existing function; however, the existing elevate is designed around mesh points. We can take care of this post PR.